### PR TITLE
Home work 4

### PR DESCRIPTION
**Pull request by FiodarKarnilovich**
AuthFilter  пока без Логата

### CODE REVIEW

File: web/pom.xml [hunk 1]
```diff
@@ -16,6 +16,35 @@
...
+        <dependency>
+            <groupId>by.karnilovich.mac</groupId>
+            <artifactId>model</artifactId>
+            <version>1.0-SNAPSHOT</version>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/jakarta.servlet/jakarta.servlet-api -->
+        <dependency>
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
+            <version>6.0.0</version>
```
Comment (anarut) [inline web/pom.xml:29]: put this version into parent properties

File: web/src/main/java/by/karnilovich/web/servlet/UserRegistrationServlet.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        final String password = req.getParameter("password");
+        final String birthDay = req.getParameter("birthDay");
+        final String phoneNumber = req.getParameter("phoneNumber");
+
+        Person person = new Person(firstName, lastName, email, password, birthDay, phoneNumber);
+        PersonUtil.addPersonToList(person);
+
+        LOGGER.info("User '{}' added in app", person.getEmail());
+        session.setAttribute(LOGGED_IN_USER, person);
+        resp.sendRedirect("/web/jsp/index_logged.jsp");
```
Comment (anarut) [inline web/src/main/java/by/karnilovich/web/servlet/UserRegistrationServlet.java]: usually registration process is not the same as login process. I think after successful registration you redirect to login page to proceed with it.

File: web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        LOGGER.debug("--- auth filter  ---");
+        {
+            HttpServletRequest req = (HttpServletRequest) servletRequest;
+            HttpServletResponse res = (HttpServletResponse) servletResponse;
+            HttpSession session = req.getSession();
+            String requestURI = req.getRequestURI();
+
+
+            if (requestURI.equals("/web/jsp/index.jsp") || requestURI.equals("/web/jsp/contacts.jsp") ||
```
Comment (anarut) [inline web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java]: all Strings in this file would be great to put into java constants

File: web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                    requestURI.equals("/web/jsp/show_list_cars.jsp") || requestURI.equals("/web/jsp/login.jsp") ||
+                    requestURI.equals("/web/jsp/user_registration.jsp")  ) {
+                LOGGER.debug(" 1 if -  "+requestURI);
```
Comment (anarut) [inline web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java]: put more effort and leave valuable comments. E.g. 
LOGGER.debug("Authentication is not required for :  " + requestURI + ";");
or
LOGGER.debug("Authentication required for :  " + requestURI + ";");
some code
LOGGER.debug("User 'john doe' is not authenticated. Redirecting to login page");

File: web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                filterChain.doFilter(servletRequest, servletResponse);
+            } else if ((requestURI.equals("/web/jsp/index_logged.jsp") || requestURI.equals("/web/jsp/view_user_details.jsp")
```
Comment (anarut) [inline web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java]: You don't need to have this list of URIs. It's enough to have list of URIs w/o auth. All other URIs require auth.

File: web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                    || requestURI.equals("/web/login") || requestURI.equals("/web/userregistration") ) ||
+                        ((session != null) && (session.getAttribute("LOGGED_IN_USER") != null))) {
+                LOGGER.debug(" 2 if "+requestURI);
+                filterChain.doFilter(servletRequest, servletResponse);
+                } else if (requestURI.equals("/web/logout")) {
```
Comment (anarut) [inline web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java]: should not be here. auth is not responsible for it (Logout servlet is)

File: web/src/main/java/by/karnilovich/web/servlet/LoginServlet.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        } else {
+            Person person = PersonUtil.getPersonList().stream()
+                    .filter(p -> p.getEmail().equalsIgnoreCase(email))
+                    .filter(p -> p.getPassword().equals(password))
+                    .findFirst().orElse(null);
+            if (person == null) {
+                writer.println("Invalid credentials. Please, Enter login and password again.");
+            } else {
+                LOGGER.info("User '{}' logged in into app", person.getEmail());
+                writer.println("All right");
```
Comment (anarut) [inline web/src/main/java/by/karnilovich/web/servlet/LoginServlet.java]: remove this line.

File: web/src/main/java/by/karnilovich/web/servlet/NewServlet.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+//import jakarta.servlet.ServletException;
+//import jakarta.servlet.annotation.WebServlet;
+//import jakarta.servlet.http.HttpServlet;
+//import jakarta.servlet.http.HttpServletRequest;
+//import jakarta.servlet.http.HttpServletResponse;
+//
+//import java.io.IOException;
+//
+//@WebServlet
+//public class NewServlet extends HttpServlet {
```
Comment (anarut) [inline web/src/main/java/by/karnilovich/web/servlet/NewServlet.java]: remove this class if you don't need it

**Review by anarut — CHANGES_REQUESTED** [review]

File: web/src/main/java/by/karnilovich/web/servlet/NewServlet.java (hunk 8 above)
Comment (FiodarKarnilovich) [reply web/src/main/java/by/karnilovich/web/servlet/NewServlet.java]: delete

File: web/pom.xml (hunk 1 above)
Comment (FiodarKarnilovich) [reply web/pom.xml:29]: added dependancy

File: web/src/main/java/by/karnilovich/web/servlet/UserRegistrationServlet.java (hunk 2 above)
Comment (FiodarKarnilovich) [reply web/src/main/java/by/karnilovich/web/servlet/UserRegistrationServlet.java]: сделал

File: web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java (hunk 4 above)
Comment (FiodarKarnilovich) [reply web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java]: пока частично

File: web/src/main/java/by/karnilovich/web/servlet/LoginServlet.java (hunk 7 above)
Comment (FiodarKarnilovich) [reply web/src/main/java/by/karnilovich/web/servlet/LoginServlet.java]: удалил

File: web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java (hunk 3 above)
Comment (FiodarKarnilovich) [reply web/src/main/java/by/karnilovich/web/servlet/AuthFilter.java]: сделал

**Review by anarut — APPROVED** [review]

**Pull request merged**